### ISSUE TITLE
Element mismatch DeclarationResolver.resolve

### ISSUE THREAD

**Issue by rtolar** [dart-lang/sdk#34721]
Received the following error while editing code:
==================
Analyzer Feedback from IntelliJ

# Version information

- `IDEA AI-181.5540.7.32.5014246`
- `2.1.0-edge.8a919426f0e8665a0e6b01c2fcf3113587f8b040`
- `AI-181.5540.7.32.5014246, JRE 1.8.0_152-release-1136-b06x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.13.6 unknown, screens 1680x1050  Retina`

# Exception

Please append the contents of:
file:///private/var/folders/kr/_0rvhwj9417dsnf42d8s43lc0000gn/T/
[report.txt](https://github.com/dart-lang/sdk/files/2458256/report.txt)
report.txt

```
Dart analysis server, SDK version 2.1.0-edge.8a919426f0e8665a0e6b01c2fcf3113587f8b040, server version 1.20.5, error: Analysis failed: /Users/richardtolar/dev/flutter_chart_example_20181008/lib/bloc.dart
Element mismatch in /Users/richardtolar/dev/flutter_chart_example_20181008/lib/bloc.dart at class RealtimeDataBloc
#0      DeclarationResolver.resolve (package:analyzer/src/generated/declaration_resolver.dart:54:7)
#1      LibraryAnalyzer._resolveFile (package:analyzer/src/dart/analysis/library_analyzer.dart:576:31)
#2      LibraryAnalyzer._analyze.<anonymous closure> (package:analyzer/src/dart/analysis/library_analyzer.dart:104:9)
#3      __InternalLinkedHashMap


===========================

**Comment by scheglov** [comment]
Looking into `report.txt`, I see that this is most probably the already fixed in 765a78d44e7a9e5fd09b10a3b0c60c6e2e4bba96.

```dart
Caused by RangeError (index): Invalid value: Not in range 0..7, inclusive: 8
#0      List.[] (dart:core/runtime/libarray.dart:12:52)
#1      _UnitResynthesizer.getReferenceInfo (package:analyzer/src/summary/resynthesize.dart:1456:43)
#2      _UnitResynthesizer.buildType (package:analyzer/src/summary/resynthesize.dart:1362:38)
#3      SummaryResynthesizerContext.resolveTypeRef (package:analyzer/src/summary/resynthesize.dart:459:30)
#4      ExecutableElementImpl.returnType (package:analyzer/src/dart/element/element.dart:4008:64)
```

**Comment by devoncarew** [comment]
Fantastic, thanks for looking into this. I expect that many people are hitting this from Flutter, so the fix for this then should be in the current Flutter on master.
